viewpump/src/main/java/io/github/inflationx/viewpump/ViewPump.java

```diff
@@ -69,6 +64,11 @@ public boolean isCustomViewCreation() {
     public static Builder builder() {
         return new Builder();
     }
+    
+    /** Returns an immutable copy of {@code list}. */
+    private static <T> List<T> immutableList(List<T> list) {
+        return Collections.unmodifiableList(new ArrayList<>(list));
```

Comment:
Does wrapping with the `unmodifiableList` not create a copy? Just wondering if you need to call the new ArrayList here?